kolibri/core/assets/src/core-actions.js

```diff
@@ -83,6 +83,78 @@ function _sessionState(data) {
   return state;
 }
 
+const masteryLoggingMap = {
+  id: 'pk',
+  start_timestamp: 'start_timestamp',
+  completion_timestamp: 'completion_timestamp',
+  end_timestamp: 'end_timestamp',
+  mastery_criterion: 'mastery_criterion',
+  complete: 'complete',
+  responsehistory: 'responsehistory',
+};
+
+function _masteryLoggingState(data) {
+  const state = {};
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    state[key] = data[masteryLoggingMap[key]];
+  });
+  return state;
+}
+
+function _masteryLogModel(store) {
+  const mapping = {};
+  const masteryLog = store.state.core.logging.mastery;
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    mapping[masteryLoggingMap[key]] = masteryLog[key];
+  });
+  mapping.summarylog = store.state.core.logging.summary.id;
+  return mapping;
+}
+
+const attemptLoggingMap = {
+  id: 'pk',
+  item: 'item',
+  start_timestamp: 'start_timestamp',
+  completion_timestamp: 'completion_timestamp',
+  end_timestamp: 'end_timestamp',
+  time_spent: 'time_spent',
+  complete: 'complete',
+  correct: 'correct',
+  answer: 'answer',
+  simple_answer: 'simple_answer',
+  interaction_history: 'interaction_history',
+};
+
+const attemptLoggingJSONifyKeys = {
+  answer: true,
+  interaction_history: true,
+};
```

Comment:
This could be good yes. One caveat would be that we would have to change the dirty checking logic for the fields that are then nested Javascript objects. (otherwise we risk either not saving, or always saving, the parts that risk being the bulkiest of the representation).